src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -141,17 +141,36 @@ private static int ComputeSymbolMatchPriority(ISymbol symbol)
             var name = SymbolCompletionItem.GetSymbolName(item);
             var kind = SymbolCompletionItem.GetKind(item);
             var isGeneric = SymbolCompletionItem.GetSymbolIsGeneric(item);
-            var relatedDocumentIds = document.Project.Solution.GetRelatedDocumentIds(document.Id).Concat(document.Id);
             var options = document.Project.Solution.Workspace.Options;
-            var totalSymbols = await base.GetPerContextSymbolsAsync(document, position, options, relatedDocumentIds, preselect: false, cancellationToken: cancellationToken).ConfigureAwait(false);
-            foreach (var (documentId, syntaxContext, symbols) in totalSymbols)
+            var relatedDocumentIds = document.Project.Solution.GetRelatedDocumentIds(document.Id);
```

Comment:
Comparing with the original implementation(...Concat(document.Id)), would 'relatedDocumentIds' contain the original document? Would it affect some corner cases?

Reply:
`Solution.GetRelatedDocumentIds` actually include the original document id in returned list.

http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/Solution.cs,1655